nvflare/fuel/f3/drivers/aio_http_driver.py

```diff
@@ -0,0 +1,184 @@
+
+log = logging.getLogger(__name__)
+
+THREAD_POOL_SIZE = 8
+MAX_MSG_SIZE = 2 * 1024 * 1024 * 1024  # 2GB
+
+
+class WsConnection(Connection):
+    def __init__(self, websocket: Any, aio_context: AioContext, connector: ConnectorInfo, secure: bool):
+        super().__init__(connector)
+        self.websocket = websocket
+        self.aio_context = aio_context
+        self.closing = False
+        self.secure = secure
+        self.conn_props = self._get_socket_properties()
+
+    def get_conn_properties(self) -> dict:
+        return self.conn_props
+
+    def close(self):
+        self.closing = True
+        self.aio_context.run_coro(self.websocket.close())
+
+    def send_frame(self, frame: BytesAlike):
+        self.aio_context.run_coro(self._async_send_frame(frame))
+
+    def _get_socket_properties(self) -> dict:
+        conn_props = {}
+
+        addr = self.websocket.remote_address
+        if addr:
+            conn_props[DriverParams.PEER_ADDR.value] = f"{addr[0]}:{addr[1]}"
+
+        addr = self.websocket.local_address
+        if addr:
+            conn_props[DriverParams.LOCAL_ADDR.value] = f"{addr[0]}:{addr[1]}"
+
+        peer_cert = self.websocket.transport.get_extra_info("peercert")
+        if peer_cert:
+            cn = get_certificate_common_name(peer_cert)
+        else:
+            if self.secure:
+                cn = "N/A"
+            else:
+                cn = None
+
+        if cn:
+            conn_props[DriverParams.PEER_CN.value] = cn
+
+        return conn_props
+
+    async def _async_send_frame(self, frame: BytesAlike):
+        try:
+            await self.websocket.send(frame)
+            # This is to yield control. See bug: https://github.com/aaugustin/websockets/issues/865
+            await asyncio.sleep(0)
+        except Exception as ex:
+            log.error(f"Error sending frame for connection {self}: {ex}")
+
+
+class AioHttpDriver(BaseDriver):
+    """Async HTTP driver using websocket extension"""
+
+    def __init__(self):
+        super().__init__()
+        self.aio_context = AioContext.get_global_context()
+        self.stop_event = None
+        self.ssl_context = None
+
+    @staticmethod
+    def supported_transports() -> List[str]:
+        return ["http", "https", "ws", "wss"]
+
+    @staticmethod
+    def capabilities() -> Dict[str, Any]:
+        return {DriverCap.HEARTBEAT.value: True, DriverCap.SUPPORT_SSL.value: True}
+
+    def listen(self, connector: ConnectorInfo):
+        self._event_loop(Mode.PASSIVE, connector)
+
+    def connect(self, connector: ConnectorInfo):
+        self._event_loop(Mode.ACTIVE, connector)
+
+    def shutdown(self):
+        self.close_all()
+
+        if self.stop_event:
+            self.stop_event.set_result(None)
+
+    @staticmethod
+    def get_urls(scheme: str, resources: dict) -> (str, str):
+        secure = resources.get(DriverParams.SECURE)
+        if secure:
+            scheme = "https"
```

Comment:
If user define the scheme as "http", but secure == True, this will implicitly change the scheme to "https". Will this cause the confusion to the user? Maybe explicitly provides the warning message to the user?

Reply:
We have to 2 ways to specify the secure connection, using "https" or "secure=True". Those are all normal use-cases. a warning probably causes more confusion.

Reply:
what happens when user set scheme to "https" and "secure" to "False"?

Reply:
It will be secure.